x/ibc/02-client/types.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"github.com/cosmos/cosmos-sdk/x/ibc/23-commitment"
+)
+
+// TODO: types in this file should be (de/)serialized with proto in the future
+// currently amkno codec handles it
+
+// ConsensusState
+type ConsensusState interface {
+	Kind() Kind
+	GetHeight() uint64
+	GetRoot() commitment.Root
+	Validate(Header) (ConsensusState, error) // ValidityPredicate
+	Equivocation(Header, Header) bool        // EquivocationPredicate
```

Comment:
On thing I've been thinking about is if byzantine evidence handling in the SDK should be it's own module, or a part of the IBC module.

As part of IBC we will need to ship detecting equivocation in the channel and counterfactual slashing and potentially generalized slashing for cross chain validation.

Just a thought at this point